src/main/java/uk/gov/companieshouse/limitedpartnershipsapi/model/PartnershipNameEnding.java

```diff
@@ -0,0 +1,30 @@
+package uk.gov.companieshouse.limitedpartnershipsapi.model;
+
+public enum PartnershipNameEnding {
+
+    LIMITED_PARTNERSHIP("Limited Partnership"),
+    LP("LP"),
+    L_DOT_P_DOT("L.P."),
+    PARTNERIAETH_CYFYNGEDIG("Partneriaeth Cyfyngedig"),
+    PC("PC"),
+    P_DOT_C_DOT("P.C.");
+
+    private final String description;
+
+    PartnershipNameEnding(String description) {
+        this.description = description;
+    };
+
+    public String getDescription() {
+        return description;
+    }
+
+    public static PartnershipNameEnding fromDescription(String description) {
```

Comment:
I think you'll need the `@JsonCreator` annotation on this method so that the JSON can be correctly marshalled to Java, when calls are made in to the API. See `BeneficialOwnersStatementType` in the ROE API for an example.

Reply:
JsonCreator may be needed when the Json is deserialised in the controller but I think that's outside the scope of this PR.